virtual_ecosystem/models/animals/animal_cohorts.py

```diff
@@ -748,3 +765,49 @@ def migrate_juvenile_probability(self) -> float:
         probability_of_dispersal = velocity / grid_side
 
         return min(1.0, probability_of_dispersal)
+
+    def inflict_non_predation_mortality(
+        self, dt: float, carcass_pool: CarcassPool
+    ) -> None:
+        """Inflict combined background, senescence, and starvation mortalities.
+
+        Args:
+            dt: The time passed in the timestep (days).
+            carcass_pool: The local carcass pool to which dead individuals go.
+
+        """
+
+        pop_size = self.individuals
+        mass_current = self.mass_current
+
+        t_to_maturity = self.time_to_maturity
+        t_since_maturity = self.time_since_maturity
+        mass_max = self.functional_group.adult_mass  # this might not be only solution
+
+        u_bg = sf.background_mortality(
+            self.constants.u_bg
+        )  # constant background mortality
+
+        if self.is_mature is True:
+            # senescence mortality is only experienced by mature adults.
+            u_se = sf.senescence_mortality(
+                self.constants.lambda_se, t_to_maturity, t_since_maturity
+            )  # senesence mortality
+        elif self.is_mature is False:
+            u_se = 0.0
+
+        u_st = sf.starvation_mortality(
+            self.constants.lambda_max,
+            self.constants.J_st,
+            self.constants.zeta_st,
+            mass_current,
+            mass_max,
+        )  # starvation mortality
+        u_t = u_bg + u_se + u_st
+
+        # Calculate the total number of dead individuals
+        # TODO: the use of ceil here might have unintended outcomes, keep an eye on it
+        number_dead = ceil(pop_size * (1 - exp(-u_t * dt)))
```

Comment:
I'm curious: what are the unexpected consequences?

`pop_size * (1 - exp(-u_t * dt))` can be, at most, `pop_size`, so there's no risk of going above the population size when using `ceil`. And I guess that, at the lower end, it makes sense that a fraction of individual dead is a full individual dead, so sort a reasonable behaviour also there, right?

Reply:
I see... 

You're already using probabilistic events in other parts of the code, so I guess it might make sense. In that case, you will have a certain number of dead individuals for sure (might be none) and 1 more that might or might not be dead. This would be rather irrelevant for ants but might make a whole difference for elephants, as you say.